src/block-management/block-holder.js

```diff
@@ -174,31 +167,56 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
 		);
 	}
 
+	getAccessibilityLabel() {
+		const { attributes, name, order, title, originalBlockTitle, getAccessibilityLabelExtra } = this.props;
+
+		let blockName = '';
+
+		if ( name === 'core/missing' ) { // is the block unrecognized?
+			blockName = title + '. ' + originalBlockTitle;
+		} else {
+			blockName = sprintf(
+				/* translators: accessibility text. %s: block name. */
+				__( '%s Block' ),
+				title, //already localized
+			);
+		}
+
+		blockName += '. ' + sprintf( __( 'Row %d.' ), order + 1 );
+
+		// Disable adding any specific block accessibility information when running e2e tests
+		if ( getAccessibilityLabelExtra ) {
+			const blockAccessibilityLabel = getAccessibilityLabelExtra( attributes );
+			blockName += blockAccessibilityLabel ? ' ' + blockAccessibilityLabel : '';
+		}
+
+		return blockName; // Use one indexing for better accessibility
+	}
+
 	render() {
-		const { isSelected, borderStyle, focusedBorderColor, isValid, name } = this.props;
+		const { isSelected, borderStyle, focusedBorderColor, isValid, title, icon } = this.props;
 
 		const borderColor = isSelected ? focusedBorderColor : 'transparent';
-		const accessibilityLabel = this.getAccessibilityLabelForBlock();
-		const blockType = getBlockType( name );
+
+		const accessibilityLabel = this.getAccessibilityLabel();
 
 		return (
 			// accessible prop needs to be false to access children
 			// https://facebook.github.io/react-native/docs/accessibility#accessible-ios-android
 			<TouchableWithoutFeedback
-				accessible={ false }
 				onPress={ this.onFocus }
 			>
-
 				<View style={ [ styles.blockHolder, borderStyle, { borderColor } ] }>
 					{ this.props.showTitle && this.renderBlockTitle() }
 					<View
-						accessibile={ true }
+						accessible={ ! isSelected }
 						accessibilityLabel={ accessibilityLabel }
+						accessibilityRole={ 'button' }
```

Comment:
This is only meant for the screenreader. Adding `accessibilityRole={ 'button' }` will translate to the screenreader saying "button" at after reading the `accessibilityLabel`. Meaning the user can interact on it like a normal button (ouble tap to select the block in this case)

Reply:
Hey @Tug 👋 I tested this and currently it is not saying "button" after reading `accessibilityLabel`. But when I move `accessibilityRole={ 'button' }` to `TouchableWithoutFeedback` level it does say it:

```
			<TouchableWithoutFeedback
				onPress={ this.onFocus }
				accessible={ ! isSelected }
				accessibilityRole={ 'button' }
			>
```

So we might want to update this like that.

Reply:
Indeed it does that on iOS when the block is unselected. I was able to confirm that moving `accessibilityRole="button"` to `TouchableWithoutFeedback`  fixes it on both platforms. However, I also I noticed that it says "button" first before reading the view's accessibility label on Android so I moved everything to `TouchableWithoutFeedback` and now it works as expected!

Reply:
It looks like CI tests are failing this time. I have an idea, we can just set an `accessibilityHint` rather than an accessibilityRole and keep the accessibilityLabel where it was. 

```
			<TouchableWithoutFeedback
				onPress={ this.onFocus }
				accessible={ ! isSelected }
				accessibilityHint={  __('Double tap to focus')  }
			>
				<View style={ [ styles.blockHolder, borderStyle, { borderColor } ] }>
					{ this.props.showTitle && this.renderBlockTitle() }
					<View
						accessibilityLabel={ accessibilityLabel }
```

Reply:
Another option would be ofc updating the CI tests according to the current settings

Reply:
Tried to debug the e2e tests but as I was wasting too much time on this I ended up reverting the change in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/937/commits/ab3f0a35ffb87d75f9e04c97a2e936f77247f9f4).
I guess it's not too critical to have "button" mentioned before the accessibility label on android. We have many improvements to do on accessibility anyway.
@pinarol could you give it a final check please :)?

Reply:
At least the issue is resolved on iOS, now I hear: "Paragraph block. Row n. Text. Button" 👍 Thanks